narwhal/consensus/src/consensus.rs

```diff
@@ -134,24 +131,31 @@ impl ConsensusState {
 
     #[allow(clippy::result_unit_err)]
     pub fn try_insert(&mut self, certificate: Certificate) -> Result<(), ()> {
-        let last_committed = self
-            .last_committed
+        Self::try_insert_in_dag(&mut self.dag, &self.last_committed, certificate)
+    }
+
+    #[allow(clippy::result_unit_err)]
+    fn try_insert_in_dag(
+        dag: &mut Dag,
+        last_committed: &HashMap<PublicKey, Round>,
+        certificate: Certificate,
+    ) -> Result<(), ()> {
+        let last_committed_round = last_committed
             .get(&certificate.origin())
```

Comment:
is the certificate's origin the authority for the last committed round for this node, or should it be the last committed round for us?